packages/utils/models.js

```diff
@@ -280,15 +363,22 @@ export const NO_PRINTER = [MODELS.MP35, MODELS.D199, MODELS.D195];
  * @returns {boolean}
  */
 export function hasPrinter() {
-  return VerifyMethodOnSystemWrapper('hasPrinter') || !_hasModelAtList(NO_PRINTER);
+  /** negação explicada e comentada em hasNoTouch() */
+  let value = VerifyMethodOnSystemWrapper('hasPrinter');
+
+  if (typeof value === 'undefined') {
+    value = !_hasModelAtList(NO_PRINTER);
+  }
+
+  return value;
 }
 
 /**
  * @description If current model have no printer
  * @returns {boolean}
  */
 export function hasNoPrinter() {
-  return !VerifyMethodOnSystemWrapper('hasPrinter') || _hasModelAtList(NO_PRINTER);
+  return !hasPrinter;
```

Comment:
```suggestion
  return !hasPrinter();
```

`hasPrinter` é um método! Logo acima

Reply:
não acredito que dei esse mole